sway-book/src/code/language/built-ins/structs/src/lib.sw

```diff
@@ -0,0 +1,53 @@
+library structs;
+
+// ANCHOR: definition
+struct Foo {
+    bar: u64,
+    baz: bool,
+}
+// ANCHOR_END: definition
+// ANCHOR: instantiation
+fn hardcoded_instantiation() {
+    // Instantiate the variable `foo` as `Foo`
+    let mut foo = Foo {
+        bar: 42,
+        baz: false,
+    };
+
+    // Access and write to "baz"
+    foo.baz = true;
+}
+
+fn variable_instantiation() {
+    // Declare variables and pass them into `Foo`
+    let number = 42;
+    let boolean = false;
+
+    let mut foo = Foo {
+        bar: number,
+        baz: boolean,
+    };
+
+    // Access and write to "baz"
+    foo.baz = true;
+}
+// ANCHOR_END: instantiation
+// ANCHOR: destructuring
+fn destructuring() {
+    let foo = Foo {
+        bar: 42,
+        baz: false,
+    };
+
+    // bar and baz are now accessible as variables
+    let Foo{bar, baz} = foo;
```

Comment:
I'm pretty sure you don't need the `Foo` here.  You should just be able to write it as `let { bar, baz } = foo;`

Reply:
This was fixed recently in the formatter, I think it just wasn't applied to `std`

Reply:
And the `Foo` does still need to be there since it is destructuring `foo`

Reply:
formatted it should look like ` let Foo { bar, baz } = foo;`